### DISCUSSION TITLE
Displaying a list of posts by category

### DISCUSSION THREAD

**Discussion by oim37** [wintercms/wn-blog-plugin#48] [Q&A] [answered]
I have several categories:
cateogry-1
cateogry-2
 - subcategory-1
 - subcategory-2
 - subcategory-3

This page knowledge-base.htm

```
title = "knowledge base"
url = "/support/kb/:slug?/:page?"
layout = "second"
is_hidden = 0

[blogCategories]
slug = "{{ :page }}"
displayEmpty = 1
categoryPage = "knowledge-base"
==
<div class="sidebar">
    {% component 'blogCategories' %}
</div>

```
**1. When i going by links, example /support/kb/cateogry-2 - page not found.**

Also i have blog page kb.htm

```
title = "Base article"
url = "/support/kb/:slug"
layout = "second"
is_hidden = 0

[blogPost]
slug = "{{ :slug }}"
categoryPage = "kb"
==
<section class="section">
	<div class="container">
	
		<div class="section_header">
		<div class="item-news-block__date">{{post.published_at|date('d.m.Y')}}</div>
			<h2 class="section_title"><b>{{ post.title }}</b></h2><br/>
			<div class="section_text">
			
			</div>
		</div>
		<div class="post">
		{{ post.content_html|raw }}<br/>
		</div>
		<a href="{{ 'novosti' | page }}" class="item-news-block__more link-line">
							{{ 'nav.all'|_ }}
		</a>
	</div>
</section>

```
And when i go to article page /support/kb/my-test-article all ok.

Also i have page with post list code

list-articles.htm

```
title = "List all"
url = "/support/kb1/:slug?/:page?"
layout = "second"
is_hidden = 0

[blogPosts]
pageNumber = "{{ :page }}"
postsPerPage = 10
noPostsMessage = "Not found"
sortOrder = "published_at desc"
categoryPage = "list-articles"
postPage = "list-articles"
==
{% for post in posts %}
{{post.published_at|date('d.m.Y')}}<br/>
<b>{{post.title}}</b><br/>
{{post.excerpt}}<br/>
<a href="{{post.url}}"><div class="item-news-block__link link-line"><u>{{ 'nav.more'|_ }}</u></div></a><br/><br/>
{% endfor %}

{% if posts.lastPage > 1 %}
 <ul class="pagination">
 {% if posts.currentPage > 1 %}
	<li><a href="{{ this.page.baseFileName | page({ (blogPosts.pageParam): (posts.currentPage-1) })}}"><u>{{ 'nav.prev'|_ }}</u></a>&nbsp;&nbsp;&nbsp;</li>
	{% endif %}
	
	{% for page in 1..posts.lastPage %}
	<li class="{{ ( posts.currentPage == page ? 'active-page' : null ) }}">
	<a href="{{ this.page.baseFineName|page({ (pageParam): page }) }}"><u>{{ page }}</u>&nbsp;</a>
		</li>
		{% endfor %}
		
	{% if posts.lastPage > posts.currentPage %}
	<li><a href="{{ this.page.baseFileName | page({ (blogPosts.pageParam): (posts.currentPage+1) }) }}">&nbsp;&nbsp;<u>{{ 'nav.next'|_ }}</u></a></li>
	{% endif %}
	</ul>
	{% endif %}

```

3. If i go to /support/kb1 i see all with pagination.

I want to make a filter so that when I go through a category, I get a list of content that relates to it, after which I can click on the more details button and read the content.

Essentially, I need to get the basic functionality of displaying content by category.

Please help. I was able to make a list of articles and when I clicked for more details, display the article itself.

I can’t navigate from categories and filter by them.

**Comment by oim37** [comment]
And {% component 'blogCategories' %} list all categories, how except 1 category, because this component use for news too. 
Component have this settings only for: 

```
[blogPosts]
exceptCategories = "news"
```

I want to except from showing list on page {% component 'blogCategories' %}

**Comment by LukeTowers** [comment, marked as answer]
Your URLs are competing with each other. Change them to the following:

**list-category-articles.htm**
```ini
url = "/support/kb/category/:slug/:page?"
```

**single-article.htm**
```ini
url = "/support/kb/article/:slug"
```

**list-articles.htm**
```ini
url = "/support/kb/articles
```

**Reply by oim37** [reply]
Thank you. It working now, 

How hide one category in {% component 'blogCategories' %}
https://github.com/wintercms/wn-blog-plugin/discussions/48#discussioncomment-6951824
